test/config/util.test.ts

```diff
@@ -40,18 +40,39 @@ describe('util', () => {
-      expect(getCommandIdPermutations('one:two').length).to.equal(numberOfPermutations('one:two'))
-      expect(getCommandIdPermutations('one:two:three').length).to.equal(numberOfPermutations('one:two:three'))
-      expect(getCommandIdPermutations('one:two:three:four').length).to.equal(numberOfPermutations('one:two:three:four'))
-      expect(getCommandIdPermutations('one:two:three:four:five').length).to.equal(numberOfPermutations('one:two:three:four:five'))
-      expect(getCommandIdPermutations('one:two:three:four:five:six').length).to.equal(numberOfPermutations('one:two:three:four:five:six'))
-      expect(getCommandIdPermutations('one:two:three:four:five:six:seven').length).to.equal(numberOfPermutations('one:two:three:four:five:six:seven'))
-      expect(getCommandIdPermutations('one:two:three:four:five:six:seven:eight').length).to.equal(numberOfPermutations('one:two:three:four:five:six:seven:eight'))
-      expect(getCommandIdPermutations('one:two:three:four:five:six:seven:eight:nine').length).to.equal(numberOfPermutations('one:two:three:four:five:six:seven:eight:nine'))
-      expect(getCommandIdPermutations('one:two:three:four:five:six:seven:eight:nine:ten').length).to.equal(numberOfPermutations('one:two:three:four:five:six:seven:eight:nine:ten'))
+    describe('returns the correct number of permutations', () => {
+      it('1', () => {
+        expect(getCommandIdPermutations('one').length).to.equal(numberOfPermutations('one'))
+      })
+      it('2', () => {
+        expect(getCommandIdPermutations('one:two').length).to.equal(numberOfPermutations('one:two'))
+      })
+      it('3', () => {
+        expect(getCommandIdPermutations('one:two:three').length).to.equal(numberOfPermutations('one:two:three'))
+      })
+      it('4', () => {
+        expect(getCommandIdPermutations('one:two:three:four').length).to.equal(numberOfPermutations('one:two:three:four'))
+      })
+      it('5', () => {
+        expect(getCommandIdPermutations('one:two:three:four:five').length).to.equal(numberOfPermutations('one:two:three:four:five'))
+      })
+      it('6', () => {
+        expect(getCommandIdPermutations('one:two:three:four:five:six').length).to.equal(numberOfPermutations('one:two:three:four:five:six'))
+      })
+      it('7', () => {
+        expect(getCommandIdPermutations('one:two:three:four:five:six:seven').length).to.equal(numberOfPermutations('one:two:three:four:five:six:seven'))
+      })
+      it('8', () => {
+        expect(getCommandIdPermutations('one:two:three:four:five:six:seven:eight').length).to.equal(numberOfPermutations('one:two:three:four:five:six:seven:eight'))
+      })
+      it('9', () => {
+        expect(getCommandIdPermutations('one:two:three:four:five:six:seven:eight:nine').length).to.equal(numberOfPermutations('one:two:three:four:five:six:seven:eight:nine'))
+      })
+      // this test takes too long to run
+      it.skip('10', () => {
```

Comment:
this was taking over 5 seconds to run, for no incremental value over `9`.  Changes to this file were attempts to improve perf and figure out where the problem is.